### PR TITLE
[Polygon Edge] Terraform AWS deployment

### PR DESCRIPTION
**Pull request by ZeljkoBenovic**
This PR adds a Get Started > Terraform AWS deployment section to Polygon Edge documentation.

### CODE REVIEW

**Review by Aleksao998 — APPROVED** [review]
Thank you for this PR. Really good article 🚀

File: docs/edge/get-started/terraform-aws-deployment.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+terraform apply
+```
+:::info
+In a production environment `terraform taint` should be run one-by-one in order to keep the blockchain network functional.
+:::
+
+## Deployment procedure
+
+### Pre deployment steps
+* clone the official terraform scripts repository
```
Comment (Aleksao998) [inline docs/edge/get-started/terraform-aws-deployment.md:94]: just a suggestion: 
clone the official polygon edge terraform scripts repository ([repo](https://github.com/aws-ia/terraform-aws-polygon-technology-edge))

I am also ok with leaving as it is, you decide and resolve comment 😃

Comment (ZeljkoBenovic) [reply docs/edge/get-started/terraform-aws-deployment.md]: Scrapped that part entirely c7a272d0c190165c2b7e286ee773e0582714f8f1.    
There is no need now to clone the repo, all the user needs to do is to import the module 😄 
Added file examples as well.

File: docs/edge/get-started/terraform-aws-deployment.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+:::info Production deployment guide
+This is the official, production ready, fully automated, AWS deployment guide.   
+
+Manual deployments to the ***[Cloud](set-up-ibft-on-the-cloud)*** or ***[Local](set-up-ibft-locally)*** 
+are recommended for testing and/or if your cloud provider is not AWS.
+:::
+
+:::info
+This deployment is PoA only.   
+If PoS mechanism is needed, just follow this ***[guide](/docs/edge/consensus/migration-to-pos)*** on now to make a switch from PoA to PoS.
```
Comment (zivkovicmilos) [inline docs/edge/get-started/terraform-aws-deployment.md:23]: Typo here, should be `on how to make a switch from PoA to PoS`

File: docs/edge/get-started/terraform-aws-deployment.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+:::
+
+This guide will, in detail, describe the process of deploying a Polygon Edge blockchain network on AWS cloud provider,
```
Comment (zivkovicmilos) [inline docs/edge/get-started/terraform-aws-deployment.md:26]: Typo here, should be `on the AWS cloud provider`

File: docs/edge/get-started/terraform-aws-deployment.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+* [aws access key ID and secret access key](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-prereqs.html#getting-started-prereqs-keys)
+
+### Terraform variables
+Three variables that must be provided, before running the deployment:
+
+* `account_id` - the AWS account ID that the Polygon Edge blockchain cluster will be deployed on.
+* `alb_ssl_certificate` - the ARN of the certificate from AWS Certificate Manager to be used by ALB for https protocol.   
+  The certificate must be generated before starting the deployment, and it must have **Issued** status.
+* `premine` - the account that will receive pre mined native currency.
+  Value must follow the official [CLI](cli-commands#genesis-flags) flag specification.
```
Comment (zivkovicmilos) [inline docs/edge/get-started/terraform-aws-deployment.md:43]: No need to put periods `.` at the end of these list items

File: docs/edge/get-started/terraform-aws-deployment.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+Only regions that have 4 availability zones are required for this deployment. Each node is deployed in a single AZ.
+
+By placing each node in a single AZ, the whole blockchain cluster is fault-tolerant to a single node (AZ) failure, as Polygon Edge implements IBFT
+consensus which allows a single node to fail in a 4 validator node cluster.
+
+### Command line access
+
+Validator nodes are not exposed in any way to the public internet (JSON-PRC is accessed only via ALB)
+and they don't even have public IP addresses attached to them.  
+Nodes command line access is possible only via [AWS Systems Manager - Session Manager](https://aws.amazon.com/systems-manager/features/).
```
Comment (zivkovicmilos) [inline docs/edge/get-started/terraform-aws-deployment.md:68]: Typo here, should be `Node command line access`

File: docs/edge/get-started/terraform-aws-deployment.md [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+
+### Base AMI upgrade
+
+This deployment uses `ubuntu-focal-20.04-amd64-server` AWS AMI. It will **not** trigger EC2 *redeployment* if the AWS AMI gets updated.
+
+If, for some reason, base AMI is required to get updated,
+it can be achieved by running `terraform taint` command for each instance, before `terraform apply`.
```
Comment (zivkovicmilos) [inline docs/edge/get-started/terraform-aws-deployment.md:75]: Typo here, should be `by running the....`

File: docs/edge/get-started/terraform-aws-deployment.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+```
+
+:::info
+In a production environment `terraform taint` should be run one-by-one in order to keep the blockchain network functional.
+:::
+
+## Deployment procedure
+
+### Pre deployment steps
+* read through [polygon-technology-edge](https://registry.terraform.io/modules/aws-ia/polygon-technology-edge/aws) terraform registry readme
```
Comment (zivkovicmilos) [inline docs/edge/get-started/terraform-aws-deployment.md:95]: Typo here, should be `read through the`

File: docs/edge/get-started/terraform-aws-deployment.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+* add `polygon-technology-edge` module to your `main.tf` file using *provision instructions* on modules readme page
```
[the remark this answers is not in the document]
Comment (zivkovicmilos) [reply docs/edge/get-started/terraform-aws-deployment.md:96]: Also, `on the modules README page`

File: docs/edge/get-started/terraform-aws-deployment.md [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+* run `terraform init` command to install all necessary Terraform dependencies
```
Comment (zivkovicmilos) [inline docs/edge/get-started/terraform-aws-deployment.md:97]: Typo here, should be `run the`

File: docs/edge/get-started/terraform-aws-deployment.md [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+* provide a new certificate in [AWS Certificate Manager](https://aws.amazon.com/certificate-manager/)
+* make sure that the provided certificate is in **Issued** state and take a note of certificate's **ARN**
```
Comment (zivkovicmilos) [inline docs/edge/get-started/terraform-aws-deployment.md:99]: Typo here, should be `is in the .. state`

Comment (zivkovicmilos) [reply docs/edge/get-started/terraform-aws-deployment.md:99]: Also, `take note of the certificate's...`

File: docs/edge/get-started/terraform-aws-deployment.md [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+
+#### `terraform.tfvars` example
+```terraform
+account_id          = "123456789"
+premine             = "0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
+alb_ssl_certificate = "arn:aws:acm:us-west-2:123456789:certificate/64c7f117-61f5-435e-878b-83186676a8af"
+```
+
+### Deployment steps
+* create `terraform.tfvars` file
```
Comment (zivkovicmilos) [inline docs/edge/get-started/terraform-aws-deployment.md:142]: Typo here, should be `create the`

File: docs/edge/get-started/terraform-aws-deployment.md [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+* once the cname record propagates, check if the chain is working properly by calling your JSON-PRC endpoint.   
+  From the example above:
+  ```shell
+    curl  https://rpc.my-awsome-blockchain.com -X POST -H "Content-Type: application/json" --data '{"jsonrpc":"2.0","method":"eth_chainId","params":[],"id":1}'
+  ```
+  
+## Destroy procedure
+:::warning
+The following procedure will permanently delete your entire infrastructure deployed with these terraform scripts.    
+Make sure you have proper [blockchain data backups](working-with-node/backup-restore) and/or you're working with testing environment.
```
Comment (zivkovicmilos) [inline docs/edge/get-started/terraform-aws-deployment.md:170]: Typo here, should be `working with a testing environment`

File: docs/edge/get-started/terraform-aws-deployment.md [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+:::
+
+If you need to remove the whole infrastructure run the following command `terraform destroy`.
```
Comment (zivkovicmilos) [inline docs/edge/get-started/terraform-aws-deployment.md:173]: Add a comma here, `If you need to remove the whole infrastructure, ...`

**Review by zivkovicmilos — APPROVED** [review]
Looks good 💯 

I've left just some minor comments.

There is a lot of work that went behind the scenes to make this guide possible, so kudos for that 🍻

File: docs/edge/get-started/terraform-aws-deployment.md (hunk 13 above)
Comment (ZeljkoBenovic) [reply docs/edge/get-started/terraform-aws-deployment.md]: fixed 42e6ab4bdb4e7c0ddb9478d1dc45f60d4dbc174e

File: docs/edge/get-started/terraform-aws-deployment.md (hunk 12 above)
Comment (ZeljkoBenovic) [reply docs/edge/get-started/terraform-aws-deployment.md]: fixed 42e6ab4bdb4e7c0ddb9478d1dc45f60d4dbc174e

File: docs/edge/get-started/terraform-aws-deployment.md (hunk 11 above)
Comment (ZeljkoBenovic) [reply docs/edge/get-started/terraform-aws-deployment.md]: fixed 42e6ab4bdb4e7c0ddb9478d1dc45f60d4dbc174e

File: docs/edge/get-started/terraform-aws-deployment.md (hunk 10 above)
Comment (ZeljkoBenovic) [reply docs/edge/get-started/terraform-aws-deployment.md]: fixed 42e6ab4bdb4e7c0ddb9478d1dc45f60d4dbc174e

File: docs/edge/get-started/terraform-aws-deployment.md (hunk 9 above)
Comment (ZeljkoBenovic) [reply docs/edge/get-started/terraform-aws-deployment.md]: fixed 42e6ab4bdb4e7c0ddb9478d1dc45f60d4dbc174e

File: docs/edge/get-started/terraform-aws-deployment.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (ZeljkoBenovic) [reply docs/edge/get-started/terraform-aws-deployment.md]: fixed 42e6ab4bdb4e7c0ddb9478d1dc45f60d4dbc174e

File: docs/edge/get-started/terraform-aws-deployment.md (hunk 7 above)
Comment (ZeljkoBenovic) [reply docs/edge/get-started/terraform-aws-deployment.md]: fixed 42e6ab4bdb4e7c0ddb9478d1dc45f60d4dbc174e

File: docs/edge/get-started/terraform-aws-deployment.md (hunk 6 above)
Comment (ZeljkoBenovic) [reply docs/edge/get-started/terraform-aws-deployment.md]: fixed 42e6ab4bdb4e7c0ddb9478d1dc45f60d4dbc174e

File: docs/edge/get-started/terraform-aws-deployment.md (hunk 5 above)
Comment (ZeljkoBenovic) [reply docs/edge/get-started/terraform-aws-deployment.md]: fixed 42e6ab4bdb4e7c0ddb9478d1dc45f60d4dbc174e

File: docs/edge/get-started/terraform-aws-deployment.md (hunk 4 above)
Comment (ZeljkoBenovic) [reply docs/edge/get-started/terraform-aws-deployment.md]: fixed 42e6ab4bdb4e7c0ddb9478d1dc45f60d4dbc174e

File: docs/edge/get-started/terraform-aws-deployment.md (hunk 3 above)
Comment (ZeljkoBenovic) [reply docs/edge/get-started/terraform-aws-deployment.md]: fixed 42e6ab4bdb4e7c0ddb9478d1dc45f60d4dbc174e

File: docs/edge/get-started/terraform-aws-deployment.md (hunk 2 above)
Comment (ZeljkoBenovic) [reply docs/edge/get-started/terraform-aws-deployment.md:23]: fixed 42e6ab4bdb4e7c0ddb9478d1dc45f60d4dbc174e

**Pull request merged**